core/templates/pages/topics-and-skills-dashboard-page/topics-and-skills-dashboard-page.controller.spec.ts

```diff
@@ -0,0 +1,199 @@
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for the topics and skills dashboard page.
+ */
+
+import { UpgradedServices } from 'services/UpgradedServices';
+
+describe('Topics and Skills Dashboard Page', function() {
+  var $scope, ctrl;
+  var $httpBackend = null;
+  var $uibModal = null;
+  var directive;
+  var TOPICS_AND_SKILLS_DASHBOARD_DATA_URL =
+      '/topics_and_skills_dashboard/data';
+  var SAMPLE_TOPIC_ID = 'hyuy4GUlvTqJ';
+  var sampleDataResults = {
+    topic_summary_dicts: [{
+      id: SAMPLE_TOPIC_ID,
+      name: 'Sample Name',
+      language_code: 'en',
+      version: 1,
+      canonical_story_count: 3,
+      additional_story_count: 0,
+      uncategorized_skill_count: 3,
+      subtopic_count: 3,
+      topic_model_created_on: 1466178691847.67,
+      topic_model_last_updated: 1466178759209.839
+    }],
+    skill_summary_dicts: [],
+    untriaged_skill_summary_dicts: [],
+    can_create_topic: true,
+    can_create_skill: true
+  };
+  beforeEach(angular.mock.module('oppia'));
+
+  beforeEach(angular.mock.module('oppia', function($provide) {
+    var ugs = new UpgradedServices();
+    for (let [key, value] of Object.entries(ugs.getUpgradedServices())) {
+      $provide.value(key, value);
+    }
+  }));
+
+  beforeEach(angular.mock.inject(function($injector, $rootScope, $controller) {
+    $httpBackend = $injector.get('$httpBackend');
+    $scope = $rootScope.$new();
+    $uibModal = $injector.get('$uibModal');
+    $httpBackend = $injector.get('$httpBackend');
+    directive = $injector.get('topicsAndSkillsDashboardPageDirective')[0];
+    ctrl = $injector.instantiate(directive.controller, {
+      $scope: $scope
+    });
+    ctrl.$onInit();
+  }));
+
+
+  it('should initialize the variables', () => {
+    const filterOptions = {sort: '',
+      keywords: '',
+      category: '',
+      status: ''};
+    expect(2).toEqual(2);
+    expect(ctrl.pageNumber).toEqual(0);
+    expect(ctrl.topicPageNumber).toEqual(0);
+    expect(ctrl.itemsPerPage).toEqual(10);
+    expect(ctrl.skillPageNumber).toEqual(0);
+    expect(ctrl.selectedIndex).toEqual(null);
+    expect(ctrl.itemsPerPageChoice).toEqual([10, 15, 20]);
+    expect(ctrl.categories).toEqual(['Mathematics']);
+    expect(ctrl.filterOptions).toEqual(filterOptions);
+    enum ESortOptions {
+      IncreasingCreatedOn = 'Newly Created',
+      DecreasingCreatedOn = 'Oldest Created',
+      IncreasingUpdatedOn = 'Most Recently Updated',
+      DecreasingUpdatedOn = 'Least Recently Updated',
+    }
+
+    enum EPublishedOptions {
+      Published = 'Published',
+      NotPublished = 'Not Published'
+    }
+    expect(ctrl.sortOptions).toEqual(ESortOptions);
+    expect(ctrl.statusOptions).toEqual(EPublishedOptions);
+  });
+
+  it('Should init the dashboard and fetch data', function() {
+    $httpBackend.expect('GET', TOPICS_AND_SKILLS_DASHBOARD_DATA_URL).respond(
+      sampleDataResults);
+    ctrl._initDashboard(false);
```

Comment:
No calling private methods from outside the file. See [frontend test guidelines](https://github.com/oppia/oppia/wiki/Frontend-unit-tests-guide#good-practices). /cc @marianazangrossi for info

Reply:
Done.